server/src/db/post_view.rs

```diff
@@ -168,11 +124,148 @@ impl PostView {
         .then_order_by(score.desc()),
     };
 
-    query = query.limit(limit).offset(offset);
+    if !show_nsfw {
+      query = query
+        .filter(nsfw.eq(false))
+        .filter(community_nsfw.eq(false));
+    };
+
+    // TODO these are wrong, bc they'll only show saved for your logged in user, not theirs
+    if saved_only {
+      query = query.filter(saved.eq(true));
+    };
+
+    if unread_only {
+      query = query.filter(read.eq(false));
+    };
+
+    PostViewQuery {
+      conn,
+      query,
+      my_user_id: None,
+      for_creator_id: None,
+      page: None,
+      limit: None,
+    }
+  }
+
+  pub fn for_community_id(mut self, for_community_id: i32) -> Self {
+    use super::post_view::post_view::dsl::*;
+    self.query = self.query.filter(community_id.eq(for_community_id));
+    self.query = self.query.then_order_by(stickied.desc());
+    self
+  }
+
+  pub fn for_community_id_optional(self, for_community_id: Option<i32>) -> Self {
+    match for_community_id {
+      Some(for_community_id) => self.for_community_id(for_community_id),
+      None => self,
+    }
+  }
+
+  pub fn for_creator_id(mut self, for_creator_id: i32) -> Self {
+    self.for_creator_id = Some(for_creator_id);
+    self
+  }
+
+  pub fn for_creator_id_optional(self, for_creator_id: Option<i32>) -> Self {
+    match for_creator_id {
+      Some(for_creator_id) => self.for_creator_id(for_creator_id),
+      None => self,
+    }
+  }
+
+  pub fn search_term(mut self, search_term: String) -> Self {
+    use super::post_view::post_view::dsl::*;
+    self.query = self.query.filter(name.ilike(fuzzy_search(&search_term)));
+    self
+  }
+
+  pub fn search_term_optional(self, search_term: Option<String>) -> Self {
+    match search_term {
+      Some(search_term) => self.search_term(search_term),
+      None => self,
+    }
+  }
+
+  pub fn url_search(mut self, url_search: String) -> Self {
+    use super::post_view::post_view::dsl::*;
+    self.query = self.query.filter(url.eq(url_search));
+    self
+  }
+
+  pub fn url_search_optional(self, url_search: Option<String>) -> Self {
+    match url_search {
+      Some(url_search) => self.url_search(url_search),
+      None => self,
+    }
+  }
+
+  pub fn my_user_id(mut self, my_user_id: i32) -> Self {
+    self.my_user_id = Some(my_user_id);
+    self
+  }
+
+  pub fn my_user_id_optional(mut self, my_user_id: Option<i32>) -> Self {
+    self.my_user_id = my_user_id;
+    self
+  }
+
+  pub fn page(mut self, page: i64) -> Self {
+    self.page = Some(page);
+    self
+  }
+
+  pub fn page_optional(mut self, page: Option<i64>) -> Self {
+    self.page = page;
+    self
+  }
```

Comment:
Actually now that I'm thinking of it, to save code, it might be better to just keep only the optional one (so as to avoid unwrap), and then if its needed anywhere just do `.for_community(Some(community_id))`

What are your thought on that to save having two methods for everything?

Reply:
I am not sure here. It would indeed remove code, but semantically speaking I think it would be weird to do `.page(Some(2))`, for example. It would also be confusing having options that take an `Option` and others that take a value
I'll let you decide.

Reply:
I'll leave as is, I'm thinking about it and its more versatile this way. thx.